src/mono/mono/mini/mini-llvm.c

```diff
@@ -12961,8 +12963,15 @@ emit_method_inner (EmitContext *ctx)
 		bb = (MonoBasicBlock*)g_ptr_array_index (bblock_list, bb_index);
 
 		// Prune unreachable mono BBs.
-		if (!(bb == cfg->bb_entry || bb->in_count > 0))
+		if (!(bb == cfg->bb_entry || bb->in_count > 0)) {
+			LLVMBasicBlockRef target_bb = ctx->bblocks [bb->block_num].call_handler_target_bb;
+			if (target_bb) {
+				LLVMBuilderRef target_builder = create_builder (ctx);
+				LLVMPositionBuilderAtEnd (target_builder, target_bb);
+				LLVMBuildUnreachable (target_builder);
```

Comment:
We always have unique `call_handler_target_bb`s? they dont' get shared?

Also at this point in emit we didn't write any other code into `target_bb`, right? so the unreachable will be the first thing in there?

Reply:
If target_bb is dead, then nothing is going to be added to call_handler_bb, so it will become invalid IR. This didn't happen before because the branch opt pass was not enabled. Will add a comment.